File: lang/java/test/fun/vac/algorithms/search/BinarySearchTest.java
import fun.vac.algorithms.search.BinarySearch;
import fun.vac.util.SequenceBuilder;
import fun.vac.util.TestRunner;

public class BinarySearchTest {

    public static boolean testBinarySearch() {
        int size = 32768;

        Integer[] arr = new Integer[size];
        SequenceBuilder.packIncreasing(arr);

        if (BinarySearch.find(arr, -1) != size) {
            return false;
        }

        if (BinarySearch.find(arr, 2_147_483_647) != size) {
            return false;
        }

        for (int i = 0; i < size; i++) {
            if (BinarySearch.find(arr, arr[i]) != i) {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {
        TestRunner.parseTest(testBinarySearch());
    }
}

File: lang/php/test/fun/vac/algorithms/search/BinarySearchTest.php
<?php
    require_once "fun/vac/algorithms/search/BinarySearch.php";
    require_once "fun/vac/util/SequenceBuilder.php";
    require_once "fun/vac/util/TestRunner.php";

    use fun\vac\algorithms\search\BinarySearch;
    use fun\vac\util\SequenceBuilder;
    use fun\vac\util\TestRunner;

    function testBinarySearch(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packIncreasing($arr);

        if (BinarySearch\find($arr, -1) != $size) {
            return false;
        }

        if (BinarySearch\find($arr, 2147483647) != $size) {
            return false;
        }

        foreach ($arr as $i => $v) {
            if (BinarySearch\find($arr, $v) != $i) {
                return false;
            }
        }

        return true;
    }

    if (count(debug_backtrace()) == 0) {
        TestRunner\parseTest(testBinarySearch());
    }
?>
